### ISSUE TITLE
createNodeMiddleware(app) requires oauth.clientId and oauth.clientSecret

### ISSUE THREAD

**Issue by stergion** [octokit/octokit.js#2211]
### Please avoid duplicates

- [X] I checked [all open bugs](https://github.com/octokit/octokit.js/issues?q=is%3Aissue+is%3Aopen+label%3Abug) and none of them matched my problem.

### Reproducible test case

```js
import { App, createNodeMiddleware } from "octokit";
import "dotenv/config";
import express from "express";

const ghApp = new App({
  appId: process.env.GITHUB_APP_ID,
  privateKey: process.env.GITHUB_PRIVATE_KEY,
  webhooks: {
    secret: process.env.GITHUB_WEBHOOK_SECRET,
  },
});

const app = express();
app.use(createNodeMiddleware(ghApp));
app.listen(3000);
```

### Please select the environment(s) that are relevant to your bug report

- [ ] TypeScript
- [ ] Enterprise
- [ ] Browsers
- [X] Node
- [ ] Deno

### Version

 octokit@1.7.1

### What happened?
I copied the code from the [octokit.js readme.md](https://github.com/octokit/octokit.js#webhooks) wanting to test github webhooks. In the example code  oauth.clientId and oauth.clientSecret are not set. Also in the [octokit/app.js Constructor](https://github.com/octokit/app.js/#constructor) oauth.clientId and oauth.clientSecret are not required.

When I try to run my code, I get the error:
throw new Error("[@octokit/app] oauth.clientId / oauth.clientSecret options are not set");

I have to say though, in [octokit/app.js Usage](https://github.com/octokit/app.js/#usage) oauth.clientId, oauth.clientSecret are provided in the code example.

Is this a documentation error or a code bug?

### Would you be interested in contributing a fix?

- [ ] yes

**Comment by RUGMJ** [comment]
Was a fix for this found? I'm having the same error now

**Comment by stergion** [comment]
I just added the `oauth: { clientId: null!, clientSecret: null! },` in the App options:

```js
const ghApp = new App({
  appId: process.env.GITHUB_APP_ID,
  privateKey: process.env.GITHUB_PRIVATE_KEY,
  webhooks: {
    secret: process.env.GITHUB_WEBHOOK_SECRET,
  },
  oauth: { clientId: null!, clientSecret: null! },
});

**Comment by gr2m** [comment]
I agree that both the `webhooks` config and the `oauth` config should be optional. I think I tried that when implementing this initially, I think the types just got so complicated that I decided not to, at least not initially. 

You should be able to just set `webhooks.secret` or `oauth.{clientId,clientSecret}` to empty strings: `""`

**Comment by didley** [comment]
I'm not sure exactly what the difference is but there's another import of `createNodeMiddleware` that doesn't require oauth to be passed. It's used in these [docs](https://docs.github.com/en/apps/creating-github-apps/writing-code-for-a-github-app/building-a-github-app-that-responds-to-webhook-events) and this [example repo](https://github.com/github/github-app-js-sample/blob/main/app.js). 

`import { createNodeMiddleware } from '@octokit/webhooks';`

I was unable to get the `octokit` import to work and was getting 504 responses. But was successful with the `@octokit/webhooks` import.

**Comment by gr2m** [comment]
`@octokit/webhooks` is the underlying library for only webhooks: https://github.com/octokit/webhooks.js. That's why it does not require OAuth settings. It also doesn't set an `octokit` instance on the context of event handlers

**Comment by corsin-ragettli** [comment]
> I agree that both the `webhooks` config and the `oauth` config should be optional. I think I tried that when implementing this initially, I think the types just got so complicated that I decided not to, at least not initially.
> 
> You should be able to just set `webhooks.secret` or `oauth.{clientId,clientSecret}` to empty strings: `""`

When I set secret to `""`, I get the following error: 
```
Error: [@octokit/webhooks] options.secret required
```
If I set the secret, I get the following error when calling the webhook:
```json
{"error":"Error: [@octokit/webhooks] signature does not match event payload and secret"}
```

So there is currently no way for me to call a webhook, Im using GHES 3.13.5

**Comment by gr2m** [comment]
If you receive webhooks, you must set a webhook secret in your GitHub App configuration and configure your code accordingly. It's security best practice and Octokit enforces it. We only discuss not to set any `webhooks` options in the `App` constructor for use cases where webhooks are not used at all.

**Comment by corsin-ragettli** [comment]
> If you receive webhooks, you must set a webhook secret in your GitHub App configuration and configure your code accordingly. It's security best practice and Octokit enforces it. We only discuss not to set any `webhooks` options in the `App` constructor for use cases where webhooks are not used at all.

I understand and agree that it should be best practice, but I cannot have webhook secrets because of the error I get (see https://github.com/octokit/octokit.js/issues/2211#issuecomment-2498069389). Was there a breaking change which is not supported by GHES 3.13.5 regarding webhook secrets?

**Comment by wolfy1339** [comment]
Please open up a new issue with all relevant details and we can help out to see why this is happening.
